### ISSUE TITLE
Can PTB Cause Java Exceptions in 2014b?

### ISSUE THREAD

**Issue by iandol** [Psychtoolbox-3/Psychtoolbox-3#236]
If I run a simple minimal PTB demo in a loop[1] on exit I get numerous Java exceptions of the following kind:

```
Exception in thread "AWT-EventQueue-0" java.lang.IllegalArgumentException: Width (0) and height (0) cannot be <= 0
t java.awt.image.DirectColorModel.createCompatibleWritableRaster(DirectColorModel.java:1016)
	at java.awt.GraphicsConfiguration.createCompatibleImage(GraphicsConfiguration.java:186)
	at sun.awt.image.SunVolatileImage.getBackupImage(SunVolatileImage.java:236)
	at sun.awt.image.VolatileSurfaceManager.getBackupSurface(VolatileSurfaceManager.java:263)
	at sun.awt.image.VolatileSurfaceManager.initialize(VolatileSurfaceManager.java:126)
	at sun.awt.image.SunVolatileImage.<init>(SunVolatileImage.java:88)
	at sun.awt.image.SunVolatileImage.<init>(SunVolatileImage.java:117)
	at java.awt.GraphicsConfiguration.createCompatibleVolatileImage(GraphicsConfiguration.java:307)
	at java.awt.GraphicsConfiguration.createCompatibleVolatileImage(GraphicsConfiguration.java:241)
	at javax.swing.RepaintManager.getVolatileOffscreenBuffer(RepaintManager.java:1030)
	at javax.swing.RepaintManager$PaintManager.paint(RepaintManager.java:1437)
	at javax.swing.RepaintManager.paint(RepaintManager.java:1249)
	at javax.swing.JComponent._paintImmediately(JComponent.java:5167)
	at javax.swing.JComponent.paintImmediately(JComponent.java:4978)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:808)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:796)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:796)
	at javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:769)
	at javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:718)
	at javax.swing.RepaintManager.access$1100(RepaintManager.java:62)
	at javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1677)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:312)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:733)
	at java.awt.EventQueue.access$200(EventQueue.java:103)
	at java.awt.EventQueue$3.run(EventQueue.java:694)
	at java.awt.EventQueue$3.run(EventQueue.java:692)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:703)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
```

I suspect that as PTB doesn't use Java to invoke OpenGL it can be responsible for this, but just wanted to check before bugging TMW...

----
[1] https://gist.github.com/iandol/a9051c1ba1fef18afef5

**Comment by kleinerm** [comment]
It doesn't use Java for OpenGL, but display reconfiguration as happens on Windows and OSX can trigger Java AWT bugs which cause memory leaks in Java which cause out-of-memory errors after some runs on some systems.

This specific problem seems to be a new problem with latest Java AWT, where our bugfix for buggy Java causes new bugs. Which OS is this? I don't know where we actually need those Java hacks. Definitely on Windows, less sure about OSX, probably not needed at all on Linux.

Try if my last commit in my kleinerm repo helps via updated PsychJavaSwingCleanup.m:
https://github.com/kleinerm/Psychtoolbox-3/commit/a517c0ac27291466c4909e73356a9b82b73b2607

**Comment by iandol** [comment]
Yes, that fixes it! Thanks Mario! This was a OS X 10.9 + Radeon HD 5870 setup I was testing for some colleagues.
